js/src/UXCapture.js

```diff
@@ -0,0 +1,78 @@
+import ExpectedMark from "./ExpectedMark";
+import View from "./View";
+
+export default class UXCapture {
+  static attachTo(window) {
+    // allow running in Node.js environment
+    if (typeof window === "undefined") {
+      throw new Error("Must provide a valid window object");
+    }
+
+    // already have UX Capture object defined, reuse it
+    if (typeof window.UX !== "undefined") {
+      return window.UX;
+    }
+
+    // assign singleton instance to window object
+    window.UX = new UXCapture(window);
+
+    return window.UX;
+  }
+
+  constructor() {
+    this.onMark = null;
+    this.onMeasure = null;
+  }
+
+  /**
+   * Sets expected marks and corresponding zones for current view
+   *
+   * @TODO re-evaluate if we should allow multiple executions of this method
+   */
+  expect(zoneConfigs) {
+    if (typeof zoneConfigs === "undefined" || !Array.isArray(zoneConfigs)) {
+      return false;
+    }
+
+    // create a view object for initial, server-side rendered page view
+    const pageView = new View({
+      onMark: mark => {
+        if (this.onMark) {
```

Comment:
if `this.onMark` and `this.onMeasure` are initialized to no-ops, you can write

```js
const pageView = new View({ onMark: this.onMark, onMeasure: this.onMeasure, zoneConfigs });
```

super concise

Reply:
I kept the closures to make sure we can call `UX.config()` after `UX.expect()` was called.

Not a big deal, but I felt that it is a useful simplification that removes dependency on sequence of execution.